### DISCUSSION TITLE
Does USER_DATABASES include databases in AG?

### DISCUSSION THREAD

**Discussion by Noiden** [olahallengren/sql-server-maintenance-solution#847] [Questions & Answers] [answered]
If I use the parameter "USER_DATABASES" on a server that has AG configured and some databases in the AG, will the Index optimize run on the databases in the AG? or only on databases that is not in AG?

Thanks.

**Comment by Stiffboard** [comment]
Indexoptimize is a read write operation only done on primary for the
AG...and databases not in AG of course

**Reply by Noiden** [reply]
Thanks for the information but my question is if I set the parameter to "USER_DATABASES" will it also include databases in the AG or must I use the parameter "AVAILABILITY_GROUP_DATABASES" for that? I'm sorry my question was a bit vague.

**Comment by Stiffboard** [comment, marked as answer]
USER_DATABASES are all available databases, meaning read write, on Primary
and databases not in AG. AVAILABILITY_GROUP_DATABASES are only AG
databases, not databases outside AG

**Reply by Stiffboard** [reply]
Documentation is crystal clear

https://ola.hallengren.com/sql-server-index-and-statistics-maintenance.html

Consider set question to answered
